packages/cfd/src/Stores/Modules/CFD/Helpers/useCfdStores.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { createContext, PropsWithChildren, useContext } from 'react';
+import { useStore } from '@deriv/stores';
+import type { TCFDStore } from 'src/types/cfd-store.types';
+
+const CFDStoreContext = createContext<TCFDStore | null>(null);
+
+export const CFDStoreProvider = ({ children }: PropsWithChildren<unknown>) => {
+    const { modules } = useStore();
+
+    return (
+        <CFDStoreContext.Provider
+            // value={memoizedValue}
+            value={modules?.cfd || {}}
+        >
+            {children}
+        </CFDStoreContext.Provider>
+    );
+};
+
+export const useCfdStore = () => {
+    const store = useContext(CFDStoreContext);
+
+    if (!store) {
+        throw new Error('useCfdStore must be used within CfdStoreContext');
```

Comment:
```suggestion
        throw new Error('useCfdStore must be used within CFDStoreProvider');
```

Reply:
<img width="657" alt="image" src="https://user-images.githubusercontent.com/129206554/233427393-3be1d1c3-da8a-4d44-a2fc-6961fcda5d07.png">

Copied this from cashier store

I think using the work `Context` is also fine though. one can get the idea about the error